Assets/MixedRealityToolkit-SDK/Features/Diagnostics/MixedRealityDiagnosticsManager.cs

```diff
@@ -0,0 +1,228 @@
+        private bool visible;
+
+        /// <inheritdoc />
+        public bool Visible
+        {
+            get
+            {
+                return visible;
+            }
+
+            set
+            {
+                if (value != visible)
+                {
+                    visible = value;
+                    DiagnosticVisualization?.SetActive(value);
+
+                    RaiseDiagnosticsChanged();
+                }
+            }
+        }
+
+        private bool showCpu;
+
+        /// <inheritdoc />
+        public bool ShowCpu
+        {
+            get
+            {
+                return showCpu;
+            }
+
+            set
+            {
+                if (value != showCpu)
+                {
+                    showCpu = value;
+                    RaiseDiagnosticsChanged();
+                }
+            }
+        }
+
+        private bool showFps;
+
+        /// <inheritdoc />
+        public bool ShowFps
+        {
+            get
+            {
+                return showFps;
+            }
+            set
+            {
+                if (value != showFps)
+                {
+                    showFps = value;
+                    RaiseDiagnosticsChanged();
+                }
+            }
+        }
+
+        private bool showMemory;
+
+        /// <inheritdoc />
+        public bool ShowMemory
+        {
+            get
+            {
+                return showMemory;
+            }
+            set
+            {
+                if (value != showMemory)
+                {
+                    showMemory = value;
+                    RaiseDiagnosticsChanged();
+                }
+            }
+        }
+
+        public GameObject DiagnosticVisualization
+        {
+            get
+            {
+                if (diagnosticVisualization != null)
+                {
+                    return diagnosticVisualization;
+                }
+
+                if (!Visible)
+                {
+                    // Don't create a gameobject if it's not needed
+                    return null;
+                }
+
+                diagnosticVisualization = GameObject.CreatePrimitive(PrimitiveType.Quad);
+                diagnosticVisualization.name = "Diagnostics";
+                diagnosticVisualization.layer = Physics.IgnoreRaycastLayer;
+
+                // Todo: position and size
+                // Todo: add text elements
+                diagnosticVisualization.AddComponent<DiagnosticBehavior>();
+
+                var component = diagnosticVisualization.GetComponent<DiagnosticBehavior>();
+                component.ShowCpu = ShowCpu;
+                component.ShowFps = ShowFps;
+                component.ShowMemory = ShowMemory;
+
+                return diagnosticVisualization;
+            }
+        }
+        #endregion IMixedRealityDiagnosticsManager
+
+        #region IMixedRealityEventSource
+        /// <inheritdoc />
+        public uint SourceId => 0;
```

Comment:
0 means it's unset. I'm not sure, but we may want to get a unique id for this somehow.

Reply:
@davidkline-ms recommended using `SourceName.GetHashCode()` as a good way to tie a unique enough value

Reply:
Sounds good to me